sources/api/netdog/src/net_config/devices/bonding.rs

```diff
@@ -0,0 +1,161 @@
+use super::validate_addressing;
+use super::{error, Dhcp4ConfigV1, Dhcp6ConfigV1, Result, RouteV1, StaticConfigV1, Validate};
+use crate::interface_name::InterfaceName;
+use crate::net_config::devices::generate_addressing_validation;
+use serde::de::Error;
+use serde::{Deserialize, Deserializer};
+use snafu::ensure;
+use std::net::IpAddr;
+
+#[derive(Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(remote = "Self")]
+pub(crate) struct NetBondV1 {
+    pub(crate) primary: Option<bool>,
+    pub(crate) dhcp4: Option<Dhcp4ConfigV1>,
+    pub(crate) dhcp6: Option<Dhcp6ConfigV1>,
+    pub(crate) static4: Option<StaticConfigV1>,
+    pub(crate) static6: Option<StaticConfigV1>,
+    #[serde(rename = "route")]
+    pub(crate) routes: Option<Vec<RouteV1>>,
+    kind: String,
+    pub(crate) mode: BondMode,
+    #[serde(rename = "min-links")]
+    pub(crate) min_links: Option<u32>,
+    #[serde(rename = "monitoring")]
+    pub(crate) monitoring_config: BondMonitoringConfig,
+    pub(crate) interfaces: Vec<InterfaceName>,
+}
+
+impl<'de> Deserialize<'de> for NetBondV1 {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let this = Self::deserialize(deserializer)?;
+        if this.kind.to_lowercase().as_str() != "bond" {
+            return Err(D::Error::custom(format!(
+                "kind of '{}' does not match 'bond'",
+                this.kind.as_str()
+            )));
+        }
+
+        Ok(this)
+    }
+}
+
+generate_addressing_validation!(&NetBondV1);
+
+impl Validate for NetBondV1 {
+    fn validate(&self) -> Result<()> {
+        validate_addressing(self)?;
+
+        let interfaces_count: u32 = self.interfaces.len() as u32;
+        ensure!(
+            interfaces_count > 0,
+            error::InvalidNetConfigSnafu {
+                reason: "bonds must have 1 or more interfaces specified"
+            }
+        );
+        if let Some(min_links) = self.min_links {
+            ensure!(
+                min_links <= interfaces_count,
+                error::InvalidNetConfigSnafu {
+                    reason: "min-links is greater than number of interfaces configured"
+                }
+            )
+        }
```

Comment:
These can be moved into custom deserializers. I have avoided moving everything in to custom derserializers due to [this serde bug](https://github.com/serde-rs/serde/issues/2157). Until we resolve this, deserialization errors are masked and therefore it's actually providing less feedback to the user. I would like to leave them for now with the understanding we will move these when we resolve the error messaging issues. This will at least result in a user seeing a specific validation failure as it is written.

Reply:
https://github.com/bottlerocket-os/bottlerocket/issues/2657 I cut this issue to cover fixing this long term. I think this improves many things, not just this particular case.